linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarGen.scala

```diff
@@ -204,31 +107,26 @@ private[emitter] final class VarGen(jsGen: JSGen, nameGen: NameGen,
   def fileLevelVarIdent(field: String, subField: String,
       origName: OriginalName = NoOriginalName)(
       implicit pos: Position): Ident = {
-    genericIdent(field, subField, origName)
+    genericIdent(field, subField :: Nil, origName)
   }
 
   def fileLevelVarIdent(field: String)(implicit pos: Position): Ident =
     fileLevelVarIdent(field, NoOriginalName)
 
   def fileLevelVarIdent(field: String, origName: OriginalName)(
       implicit pos: Position): Ident = {
-    genericIdent(field, origName)
-  }
-
-  private def genericIdent(field: String)(
-      implicit pos: Position): Ident = {
-    genericIdent(field, NoOriginalName)
-  }
-
-  private def genericIdent(field: String, origName: OriginalName)(
-      implicit pos: Position): Ident = {
-    Ident(avoidClashWithGlobalRef("$" + field), origName)
+    genericIdent(field, Nil, origName)
   }
 
-  private def genericIdent(field: String, subField: String,
+  private def genericIdent(field: String, subFieldParts: List[String],
       origName: OriginalName = NoOriginalName)(
       implicit pos: Position): Ident = {
-    Ident(avoidClashWithGlobalRef("$" + field + "_" + subField), origName)
+    if (subFieldParts.isEmpty) {
+      Ident(avoidClashWithGlobalRef("$" + field), origName)
+    } else {
+      val subField = subFieldParts.mkString("__")
```

Comment:
Have you considered pushing this computation, and the `if/else`, down inside the method of `Scope`? That might save on `List` creations and iterations, which currently seem a bit wasteful.

Reply:
Good point. I just used `""` as an empty sub-field sentinel.